java/query/query-lambda/src/main/java/sleeper/query/lambda/WarmQueryExecutorLambda.java

```diff
@@ -0,0 +1,126 @@
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static sleeper.configuration.properties.instance.CdkDefinedInstanceProperty.CONFIG_BUCKET;
+import static sleeper.configuration.properties.instance.CdkDefinedInstanceProperty.QUERY_QUEUE_URL;
+import static sleeper.configuration.properties.table.TableProperty.TABLE_NAME;
+import static sleeper.query.runner.output.NoResultsOutput.NO_RESULTS_OUTPUT;
+
+/**
+ * A lambda that is triggered when a serialised query arrives on an SQS queue. A processor executes the request using a
+ * {@link QueryExecutor} and publishes the results to either SQS or S3 based on the configuration of the query.
+ * The processor contains a cache that includes mappings from partitions to files in those partitions. This is reused by
+ * subsequent calls to the lambda if the AWS runtime chooses to reuse the instance.
+ */
+@SuppressWarnings("unused")
+public class WarmQueryExecutorLambda implements RequestHandler<ScheduledEvent, Void> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(WarmQueryExecutorLambda.class);
+
+    private Instant lastUpdateTime;
+    private InstanceProperties instanceProperties;
+    private TablePropertiesProvider tablePropertiesProvider;
+    private final AmazonSQS sqsClient;
+    private final AmazonS3 s3Client;
+    private final AmazonDynamoDB dynamoClient;
+    private QueryMessageHandler messageHandler;
+    private SqsQueryProcessor processor;
+
+    public WarmQueryExecutorLambda() throws ObjectFactoryException {
+        this(AmazonS3ClientBuilder.defaultClient(), AmazonSQSClientBuilder.defaultClient(),
+                AmazonDynamoDBClientBuilder.defaultClient(), System.getenv(CONFIG_BUCKET.toEnvironmentVariable()));
+    }
+
+    public WarmQueryExecutorLambda(AmazonS3 s3Client, AmazonSQS sqsClient, AmazonDynamoDB dynamoClient, String configBucket) throws ObjectFactoryException {
+        this.s3Client = s3Client;
+        this.sqsClient = sqsClient;
+        this.dynamoClient = dynamoClient;
+        instanceProperties = loadInstanceProperties(s3Client, configBucket);
+        tablePropertiesProvider = new TablePropertiesProvider(instanceProperties, s3Client, dynamoClient);
+    }
+
+    @Override
+    public Void handleRequest(ScheduledEvent event, Context context) {
+        List<TableProperties> tableProperties = S3TableProperties.getStore(instanceProperties, s3Client, dynamoClient)
+                    .streamAllTables().collect(Collectors.toList());
+
+        LOGGER.info("Starting to build queries for the tables");
+        tableProperties.forEach(tableProperty -> {
+
+            Schema schema = tableProperty.getSchema();
+            Field field = schema.getRowKeyFields().get(0);
+            Region region = new Region(Collections.singletonList(new Range.RangeFactory(schema)
+                    .createRange(field, "a", "aa")));
```

Comment:
This will only work if field is a string type. We need to create a range based on the type of the field.

Reply:
I think this is now fixed